lib/http_client.js

```diff
@@ -89,6 +100,13 @@ class HttpClient extends EventEmitter {
         }
     }
 
+    _setSessionCookie(cookie) {
+        if (cookie.startsWith('session=')) {
+            var value = cookie.substring(0, cookie.indexOf(';'));
```

Comment:
Is it always ending with ";"? Otherwise we would get empty string.

Reply:
Personally at least, in these kinds of cases when the data received is not under the control of the driver, i would place appropriate checks to ensure the data received is as expected and if not, throw something like NoSQLProtocolError (this also applies to the previous comment about the array).  (the only exception is if these checks were already done by the lower layer which in this case is node.js http module)   Do you see any reason not to do this?  I m fine if you want to skip these checks, but still wanted to understand the rationale for this.

Reply:
I'd rather not throw an exception in such paths because there is no reason that the application should not continue but making the checks and logging issues is a reasonable thing to do

Reply:
Ok, i see, yeah that makes sense.  Since there is no logging there yet, this should be ok then.